wgpu-core/src/device/resource.rs

```diff
@@ -1309,6 +1319,7 @@ impl Device {
                 texture_format: texture.desc.format,
                 format: resolved_format,
                 dimension: resolved_dimension,
+                usage: resolved_usage,
```

Comment:
There is one more spot above that needs to be updated:

```rust
texture.hal_usage & mask_copy & mask_dimension & mask_mip_level
```

I think we can now remove `Texture.hal_usage` and can just call `conv::map_texture_usage(resolved_usage, aspects, format_flags)`.

Reply:
I did not feel confident doing that as there is a clear distinction between public usages (TextureUsage) and private usages (`TextureUses`).

Reply:
I don't think we will behave properly if we don't. Although the only case I know of that this matters in is: If you enable `bgra8unorm-storage` and create a texture with the `bgra8unorm` format & storage+renderable usage and then want to create a `bgra8unorm-srgb` renderable view from it (what we will pass to `wgpu-hal` is `bgra8unorm-srgb` with storage+renderable usage).

We can fix this later but I figured I'd point it out.